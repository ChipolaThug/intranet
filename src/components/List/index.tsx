import { memo } from "react"
import { Link } from "react-router-dom"

import { User } from "../../services/jsonPlaceholder"
import styles from "./styles.module.scss"

interface Props {
    items: User[]
}

const List = ({ items }: Props) => (
    <div className={styles.UserList}>
        <h4>User List</h4>
        <ul>
            {items.map(({ membreId, name }) => (
                <li key={membreId}>
                    <Link to={`/UserInfo/${membreId}`}>{name}</Link>
                </li>
            ))}
        </ul>
    </div>
)

export default memo(List)
