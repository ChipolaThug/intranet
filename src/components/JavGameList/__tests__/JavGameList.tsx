/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"

import List from "../index"

describe("<List />", () => {
    it("renders", () => {
        const tree = render(
            <MemoryRouter>
                <List
                    items={[
                        {
                            id: 5,
                            titre: "6 qui prend!",
                            auteur: "Wolfgang Kramer",
                            editeur: "(uncredited) , Design Edge , B",
                            minJoueurs: 2,
                            maxJoueurs: 10,
                            duree: 45,
                            type: "Ambiance",
                            poufpaf: "0-9-2/6-qui-prend-6-nimmt",
                            photo: "https://cf.geekdo-images.com/thumb/img/lzczxR5cw7an7tRWeHdOrRtLyes=/fit-in/200x150/pic772547.jpg",
                            bggPhoto: "",
                            bggId: 432,
                            exemplaires: 1,
                            dispoPret: 1,
                            nonRangee: 0,
                            horodatage: "0000-00-00",
                            ean: "3421272101313",
                        },
                    ]}
                />
            </MemoryRouter>
        ).container.firstChild

        expect(tree).toMatchSnapshot()
    })
})
