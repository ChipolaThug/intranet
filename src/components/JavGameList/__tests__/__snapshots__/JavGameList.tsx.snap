// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders 1`] = `
<div
  class="JavGameList"
>
  <h4>
    JAV Games
  </h4>
  <ul>
    <li>
      <a
        href="/UserInfo/5"
      >
        6 qui prend!
      </a>
    </li>
  </ul>
</div>
`;
